src/nisarqa/utils/metadata_checks.py

```diff
@@ -168,102 +166,204 @@ def verify_metadata_cubes(
+        True to raise an exception if one of the metadata datasets contains
+        all non-finite (e.g. Nan, +/- Inf) values, or if one of the
+        z-dimension height layers in a 3D dataset has all non-finite values.
+        False to quiet the exception, although it will still be logged.
+        Defaults to True.
+
+    Raises
+    ------
+    nisarqa.InvalidRasterError
+        If `fail_if_all_nan` is True and if one or more metadata datasets
+        contains all non-finite (e.g. Nan, +/- Inf) values, or if one of
+        the z-dimension height layers in a 3D dataset has all non-finite values.
+    """
+    log = nisarqa.get_logger()
+    # Flag indicating if metadata datasets pass all verification checks; used
+    # for Summary CSV reporting
+    passes = True
+    has_finite = True
 
+    # Check metadata datasets in metadata Group
+    try:
+        # Note: During the __post_init__ of constructing each metadata dataset,
+        # several validation checks are performed, including ensuring that
+        # there are corresponding datasets with x coordinates and y coordinates
+        # of the correct length. If these elements are missing, exceptions
+        # will get thrown.
         spec = nisarqa.Version.from_string(product.product_spec_version)
 
-        # Non-InSAR products have calibrationInformation groups
-        if isinstance(product, nisarqa.NonInsarProduct):
-            for freq in product.freqs:
-
-                if spec >= nisarqa.Version(1, 1, 0):
-                    for cube in product.neb_metadata_cubes(freq):
-                        has_finite &= _metadata_cube_has_finite_pixels(cube)
-                        passes &= has_finite
-                        passes &= _metadata_cube_has_finite_pixels(cube)
-                        passes &= _metadata_cube_is_not_all_zeros(cube)
-                        passes &= _check_gdal(c=cube)
-
-                    for cube in product.elevation_antenna_pat_metadata_cubes(
-                        freq
-                    ):
-                        has_finite &= _metadata_cube_has_finite_pixels(cube)
-                        passes &= has_finite
-                        passes &= _metadata_cube_is_not_all_zeros(cube)
-                        passes &= _check_gdal(c=cube)
-
-                    if isinstance(product, nisarqa.RSLC):
-                        for cube in product.geometry_metadata_cubes():
-                            has_finite &= _metadata_cube_has_finite_pixels(cube)
-                            passes &= has_finite
-                            passes &= _metadata_cube_is_not_all_zeros(cube)
-
-                        for cube in product.crosstalk_metadata_cubes():
-                            has_finite &= _metadata_cube_has_finite_pixels(cube)
-                            passes &= has_finite
-                            passes &= _metadata_cube_is_not_all_zeros(cube)
+        for freq in product.freqs:
+
+            if spec < nisarqa.Version(1, 1, 0):
```

Comment:
Should there be a comment here explaining what's going on? Let's explain why we're skipping this step if the spec version is < 1.1.0